### ISSUE TITLE
assembly-informational-format: ignored

### ISSUE THREAD

**Issue by kevinneufeld** [GitTools/GitVersion#870]
I am trying to customize the AssemblyInformationalVersion, either with GitVersion.yml or with command line argument `/overriderconfig:assembly-informational-format="{NuGetVersion}"`

Unfortunately neither appear to be working. Maybe I am missed something in the docs?
Is there hidden argument to force an update from GitVersion.yml? 

My AssemblyInfo.cs is always updated with 
```C#
[assembly: AssemblyInformationalVersion("2.4.0-beta.1+35.Branch.release-2.4.0.Sha.edee73f7b0e1384954d0f3b1d50a1e6e772eac42")]
``` 

when I want 
 ```C#
 [assembly: AssemblyInformationalVersion("2.4.0-beta0001")]
 ``` 

**GitVersion.yml:** (located in the root of the repo)

```yml
assembly-informational-format: '{NuGetVersionV2}'
mode: ContinuousDeployment
branches: {}
ignore:
  sha: []
```

Using the following commands: 

```
  PS> GitVersion /updateassemblyinfo
  PS> GitVersion /overrideconfig:assembly-informational-format="{NuGetVersion}" /updateassemblyinfo
```
**GitVersion /showconfig**
```
  PS> gitversion /showconfig
      assembly-versioning-scheme: MajorMinorPatch
      mode: ContinuousDelivery
      tag-prefix: '[vV]'
      continuous-delivery-fallback-tag: ci
      major-version-bump-message: '\+semver:\s?(breaking|major)'
      minor-version-bump-message: '\+semver:\s?(feature|minor)'
      patch-version-bump-message: '\+semver:\s?(fix|patch)'
      legacy-semver-padding: 4
      build-metadata-padding: 4
      commits-since-version-source-padding: 4
      commit-message-incrementing: Enabled
      branches:
        master:
          mode: ContinuousDelivery
          tag: ''
          increment: Patch
          prevent-increment-of-merged-branch-version: true
          track-merge-target: false
        releases?[/-]:
          mode: ContinuousDelivery
          tag: beta
          increment: Patch
          prevent-increment-of-merged-branch-version: true
          track-merge-target: false
        features?[/-]:
          mode: ContinuousDelivery
          tag: useBranchName
          increment: Inherit
          prevent-increment-of-merged-branch-version: false
          track-merge-target: false
        (pull|pull\-requests|pr)[/-]:
          mode: ContinuousDelivery
          tag: PullRequest
          increment: Inherit
          prevent-increment-of-merged-branch-version: false
          tag-number-pattern: '[/-](?<number>\d+)[-/]'
          track-merge-target: false
        hotfix(es)?[/-]:
          mode: ContinuousDelivery
          tag: beta
          increment: Patch
          prevent-increment-of-merged-branch-version: false
          track-merge-target: false
        support[/-]:
          mode: ContinuousDelivery
          tag: ''
          increment: Patch
          prevent-increment-of-merged-branch-version: true
          track-merge-target: false
        dev(elop)?(ment)?$:
          mode: ContinuousDeployment
          tag: unstable
          increment: Minor
          prevent-increment-of-merged-branch-version: false
          track-merge-target: true
      ignore:
        sha: []
  ```

**Comment by kevinneufeld** [comment]
Work around, delete the folder and/or contents: `.git\gitversion_cache\*`.

Would be nice to have a command line option to force (delete cache) update.

**Comment by asbjornu** [comment]
@kevinneufeld What we obviously need to do is expand upon #808 and include the command line argument string in the cache key as well. Are you up for submitting a PR for that?

**Comment by sergiorykov** [comment]
It's caused by incorrect cache key computation. It uses obsolete config file name `GitVersionConfig.yaml`. See details in #863.

**Comment by johncrim** [comment]
+1 Thank you - it appeared that using config options was broken, but the workaround `rm -r .\.git\gitversion_cache` solved the problem.

**Comment by JakeGinnivan** [comment]
Fixed in 3.6.1

**Comment by brunobertechini** [comment]
Maybe is the same thing but I got an exception when using GitVersion.yml and running GitVersion.exe (and also same exception on VSTS Team Build Server):

This is the content of GitVersion.yml
`assembly-informational-version: '{NugetVersion}'`

Exception:
```
INFO [09/27/16 18:29:37:45] Working directory: C:\My\Path\To\Repository
INFO [09/27/16 18:29:37:46] IsDynamicGitRepository: False
INFO [09/27/16 18:29:37:49] Returning Project Root from DotGitDirectory: C:\My\Path\To\Repository\.git - C:\My\Path\To\Repository
INFO [09/27/16 18:29:37:49] Running on Windows.
INFO [09/27/16 18:29:37:49] IsDynamicGitRepository: False
INFO [09/27/16 18:29:37:49] Returning Project Root from DotGitDirectory: C:\My\Path\To\Repository\.git - C:\My\Path\To\Repository
INFO [09/27/16 18:29:37:49] Project root is: C:\My\Path\To\Repository
INFO [09/27/16 18:29:37:49] DotGit directory is: C:\My\Path\To\Repository\.git
INFO [09/27/16 18:29:37:50] IsDynamicGitRepository: False
INFO [09/27/16 18:29:37:50] Returning Project Root from DotGitDirectory: C:\My\Path\To\Repository\.git - C:\My\Path\To\Repository
INFO [09/27/16 18:29:37:56] Begin: Loading version variables from disk cache
  INFO [09/27/16 18:29:37:57] Cache file C:\My\Path\To\Repository\.git\gitversion_cache\3F67CE2916EE2FC1D954442039D9D3781D32BCD9.yml not found.
INFO [09/27/16 18:29:37:57] End: Loading version variables from disk cache (Took: 2.00ms)
INFO [09/27/16 18:29:37:57] IsDynamicGitRepository: False
INFO [09/27/16 18:29:37:57] Returning Project Root from DotGitDirectory: C:\My\Path\To\Repository\.git - C:\My\Path\To\Repository
ERROR [09/27/16 18:29:37:64] An unexpected error occurred:
YamlDotNet.Core.YamlException: (Line: 1, Col: 1, Idx: 0) - (Line: 1, Col: 1, Idx: 0): Exception during deserialization ---> System.Runtime.Serialization.SerializationException: Property 'assembly-informational-version' not found on type 'GitVersion.Config'.
   em YamlDotNet.Serialization.TypeInspectors.TypeInspectorSkeleton.GetProperty(Type type, Object container, String name, Boolean ignoreUnmatched)
   em YamlDotNet.Serialization.NodeDeserializers.ObjectNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(EventReader reader, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
   em YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(EventReader reader, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   --- Fim do rastreamento de pilha de exceções internas ---
   em YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(EventReader reader, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   em YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(EventReader reader, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   em YamlDotNet.Serialization.Deserializer.Deserialize(EventReader reader, Type type)
   em YamlDotNet.Serialization.Deserializer.Deserialize[T](TextReader input)
   em GitVersion.ConfigSerialiser.Read(TextReader reader)
   em GitVersion.ConfigurationProvider.Provide(String workingDirectory, IFileSystem fileSystem, Boolean applyDefaults, Config overrideConfig)
   em GitVersion.ExecuteCore.ExecuteInternal(String targetBranch, String commitId, GitPreparer gitPreparer, IBuildServer buildServer, Config overrideConfig)
   em GitVersion.ExecuteCore.ExecuteGitVersion(String targetUrl, String dynamicRepositoryLocation, Authentication authentication, String targetBranch, Boolean noFetch, String workingDirectory, String commitId, Config overrideConfig)
   em GitVersion.SpecifiedArgumentRunner.Run(Arguments arguments, IFileSystem fileSystem)
   em GitVersion.Program.VerifyArgumentsAndRun()
INFO [09/27/16 18:29:37:64]
INFO [09/27/16 18:29:37:64] Attempting to show the current git graph (please include in issue):
INFO [09/27/16 18:29:37:64] Showing max of 100 commits
INFO [09/27/16 18:29:37:84] * 55d9a41 13 minutes ago  (HEAD -> develop, origin/develop)
* 8541f5c 15 minutes ago
* fc70d4e 19 minutes ago
* 6ea8134 31 minutes ago
* a8c4dca 64 minutes ago
* 50084c7 68 minutes ago
* 539f8b2 72 minutes ago
* a3b256a 83 minutes ago
* 1eaaa17 5 hours ago
* a2bcf0c 2 days ago
* 8ae274e 2 days ago
* 32134b0 2 days ago
* 80b728d 2 days ago
* 10adcc1 2 days ago
* db9637e 2 days ago
* 6919df6 2 days ago
* 9cecb89 2 days ago
* abc4b00 2 days ago
* 038792f 2 days ago
| *   844a110 2 days ago  (refs/stash)
| |\
|/ /
| * ce0779c 2 days ago
|/
* f315dbf 2 days ago  (origin/master, master)
```
